pkg/controller/agent/vlanconfig/controller.go

```diff
@@ -73,22 +73,24 @@ func (h Handler) OnChange(key string, vc *networkv1.VlanConfig) (*networkv1.Vlan
 	}
 	klog.Infof("vlan config %s has been changed, spec: %+v", vc.Name, vc.Spec)
 
-	ok, v, err := h.MatchNode(vc)
+	isMatched, err := h.MatchNode(vc)
 	if err != nil {
 		return nil, err
 	}
-	// Not match
-	if !ok {
-		if err := h.removeVLAN(vc); err != nil {
-			return nil, err
-		}
-		return vc, nil
+	isApplied, err := h.isApplied(vc)
+	if err != nil {
+		return nil, err
 	}
-	// Another vlanConfig has take effect on this node
-	if v != "" && v != vc.Name {
-		klog.Infof("vlanConfig %s matches the node %s but goes after %s", vc.Name, h.nodeName, v)
+
+	if !isMatched {
+		if isApplied {
+			if err := h.removeVLAN(vc); err != nil {
+				return nil, err
+			}
+		}
 		return vc, nil
 	}
+
```

Comment:
Last question:

`OnChange` may trigger `setupVLAN` mutli times, how to avoid duplicated set, can `isApplied` be helpful?

Reply:
Yes, it doesn't matter to call `setupVLAN` multiple times. `isApplied` is used to confirm whether the vlanconfig is applied into the node, but it's powerless to check whether is required to setup VLAN again.